src/lib.rs

```diff
@@ -671,17 +700,50 @@ impl<K: EnrKey> Enr<K> {
         self.set_socket(socket, key, false)
     }
 
+    /// Unsets the `ip` and `udp` fields on the ENR.
+    pub fn remove_udp_socket(&mut self, key: &K) -> Result<(), Error> {
+        let keys_to_remove = [IP_ENR_KEY, UDP_ENR_KEY].iter();
+        let keys_to_insert = std::iter::empty::<(Vec<u8>, &[u8])>();
+        self.remove_insert(keys_to_remove, keys_to_insert, key)
+            .map(|_| ())
+    }
+
+    /// Unsets the `ip6` and `udp6` fields on the ENR.
+    pub fn remove_udp6_socket(&mut self, key: &K) -> Result<(), Error> {
+        let keys_to_remove = [IP6_ENR_KEY, UDP6_ENR_KEY].iter();
+        let keys_to_insert = std::iter::empty::<(Vec<u8>, &[u8])>();
+        self.remove_insert(keys_to_remove, keys_to_insert, key)
+            .map(|_| ())
+    }
+
     /// Sets the IP and TCP port in a single update with a single increment in sequence number.
     pub fn set_tcp_socket(&mut self, socket: SocketAddr, key: &K) -> Result<(), Error> {
         self.set_socket(socket, key, true)
     }
 
+    /// Unsets the `ip` and `tcp` fields on the ENR.
+    pub fn remove_tcp_socket(&mut self, key: &K) -> Result<(), Error> {
+        let keys_to_remove = [IP_ENR_KEY, TCP_ENR_KEY].iter();
+        let keys_to_insert = std::iter::empty::<(Vec<u8>, &[u8])>();
+        self.remove_insert(keys_to_remove, keys_to_insert, key)
+            .map(|_| ())
+    }
+
+    /// Unsets the `ip6` and `tcp6` fields on the ENR.
+    pub fn remove_tcp6_socket(&mut self, key: &K) -> Result<(), Error> {
+        let keys_to_remove = [IP6_ENR_KEY, TCP6_ENR_KEY].iter();
+        let keys_to_insert = std::iter::empty::<(Vec<u8>, &[u8])>();
+        self.remove_insert(keys_to_remove, keys_to_insert, key)
+            .map(|_| ())
+    }
+
     /// Helper function for `set_tcp_socket()` and `set_udp_socket`.
     fn set_socket(&mut self, socket: SocketAddr, key: &K, is_tcp: bool) -> Result<(), Error> {
+        let enr_backup = self.clone();
```

Comment:
nit: wdyt of doing the opposite:
```suggestion
        let new_enr = self.clone();
```
call `new_enr.seq.checked_add(1)`

and only assigning `self = new_enr;` before returning `Ok(())`

that way if in the future we need more changes that imply mutability we only change `self` once in the success case instead of multiple error situations.